File: catalog/backends/ruby-rails/patterns/unindexed_query_mongo_repository.rb
require 'mongo'

# Pattern: DBM unindexed query — collection scan (COLLSCAN)
# Adapt: replace collection and field names with domain entities
MONGO_URL = ENV.fetch("MONGODB_URL", "mongodb://demo:demo@mongodb:27017/demo")

def self.find_by_category(category)
  client = Mongo::Client.new(MONGO_URL)
  client[:events].find(category: category).sort(created_at: -1).limit(50).to_a
end

File: catalog/backends/php-laravel/patterns/UnindexedQueryMongoRepository.php
<?php

// Pattern: DBM unindexed query — collection scan (COLLSCAN)
// Adapt: replace collection and field names with domain entities

function findByCategory(string $category): array
{
    $manager = new MongoDB\Driver\Manager(env('MONGODB_URL', 'mongodb://demo:demo@mongodb:27017/demo'));
    $query = new MongoDB\Driver\Query(
        ['category' => $category],
        ['sort' => ['created_at' => -1], 'limit' => 50]
    );
    return $manager->executeQuery('demo.events', $query)->toArray();
}
